### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by rjmarin**
Complete:
-Fix submission 1
-Reblock of a model block 
-test of Reblock
Ready to master!
**Sorry for the name of the last commit! **

### CODE REVIEW

File: src/controllers/block.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ def block_attribute(self, attribute):
         if attribute not in self.block_columns:
             return("Not valid attribute")
         return_attributes = {}
-        for key in self.to_json().keys():
-            if key != attribute and key != "_id":
-                return_attributes[key] = self.to_json()[key]
+        # for key in self.to_json().keys():
```
Comment (alejandroechev) [inline src/controllers/block.py]: Remove commented code (thats one of the benefits of using Git, you can delete code, an still look it up in the previous version if you need it)

File: src/controllers/blockModel.py [hunk 2]
```diff
@@ -19,3 +24,94 @@ def find_block(self, x, y, z, collection):
...
+            for y in range(0, y_max, ry):
+                position = [position[0], position[1] + 1, -1]
+                for z in range(0, z_max, rz):
+                    position = [position[0], position[1], position[2] + 1]
+                    block = self.resize(collection, x, y, z, rx, ry, rz,position, counter, attributes_types, mass_attribute)
+                    reblock_collection.insert_one(block)
+                    counter += 1
+        return reblock_collection
+
+    def continue_attributes(self, blocks, attribute):
```
Comment (alejandroechev) [inline src/controllers/blockModel.py]: typo: continuous

File: src/controllers/blockModel.py [hunk 3]
```diff
@@ -19,3 +24,94 @@ def find_block(self, x, y, z, collection):
...
+    def categorical_attributes(self, blocks, attribute):
+        attributes = []
+        if len(blocks) == 0:
+            return None
+        for block in blocks:
+            attributes.append(block[attribute])
+        return max(set(attributes), key = attributes.count) 
+
+
+    def max_coordenates(self, collection):
```
Comment (alejandroechev) [inline src/controllers/blockModel.py]: typo: coordinates

File: src/controllers/blockModel.py [hunk 4]
```diff
@@ -19,3 +24,94 @@ def find_block(self, x, y, z, collection):
...
+        max_x = collection.find({}).sort([('x',-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["x"]
+        max_y = collection.find({}).sort([("y",-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["y"]
+        max_z = collection.find({}).sort([("z",-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["z"]
+        return [max_x, max_y, max_z]
+
+
+    def resize(self, collection, x_start, y_start, z_start ,reblock_x, reblock_y, reblock_z, position, counter, attributes_types, mass_attribute):
```
Comment (alejandroechev) [inline src/controllers/blockModel.py]: This is a confusing name, you should rename it to something like "`create_reblocked_block`" or similar

**Review by alejandroechev — CHANGES_REQUESTED** [review]

**Review by alejandroechev — CHANGES_REQUESTED** [review]
Your tests for Block class should be improved in the following way: for one method, you should have separate tests that assert only one specific case you are testing, instead of only one test with many asserts. This way you can specify in the test what is the particular case you are testing (example: check that the expected value of grade is returned for type 1, etc)

 Other improtant test cases for reblock, is how the reblock factor is used. For example what happens if the reblock factor is not divisor of the block model size. You should have different tests for this for each of the dimensions (x,y,z), again, with the goal of giving you more specific information.

In general a good skill o acquire is to try to define many tests that test different specific case for the algorithm. This will help you find possible bugs, but also help you reason about the algorithm and find edge cases where it can fails.

**Review by alejandroechev — APPROVED** [review]
